### ISSUE TITLE
Package not found by packagist (Installation error)

### ISSUE THREAD

**Issue by carloscarucce** [guzzle/guzzle#2451]
**Guzzle version(s) affected**: ^5.3.3|^6.2.1

**Description**  
The package can't be found by composer. 
I have tried to clear composer cache and update it to it's last version but the problem persists.
Here is the error message thrown:

```
Problem 1                                                                                                    
      - Installation request for aws/aws-sdk-php 3.* -> satisfiable by aws/aws-sdk-php[3.128.2].               
      - aws/aws-sdk-php 3.128.2 requires guzzlehttp/guzzle ^5.3.3|^6.2.1 -> no matching package found.
```
**How to reproduce**  
Execute composer "install" or "update" command

**Comment by sagikazarmark** [comment]
Is it possible that you have conflicting version constraints?

Can you try installing only Guzzle in an empty project?

I can't reproduce the issue.

**Comment by carloscarucce** [comment]
Maybe? There are 2 dependencies in the project I'm currently working with that depends on guzzle:

**"aws/aws-sdk-php": "3.*" (requires guzzle ver.: ^5.3.3|^6.2.1)** and **"googleads/googleads-php-lib": "37.1.0" (requires guzzle ver.: ~6.0)**

**Comment by gmponos** [comment]
Can you try the why not command of composer?

**Comment by GrahamCampbell** [comment]
Could you paste your `composer.json` file here?

**Comment by davisshaver** [comment]
This is blocking updates for me too. Here is the output of why not command.

```
aws/aws-sdk-php       3.117.2  requires  guzzlehttp/guzzle (^5.3.3|^6.2.1)  
league/oauth2-client  2.4.1    requires  guzzlehttp/guzzle (^6.0)  
```

These look compatible so I'm not sure why composer is saying conflict here. Thank you all for the help on this issue.

**Comment by GrahamCampbell** [comment]
Could you paste your composer.json file here?

**Comment by davisshaver** [comment]
I tried to recreate the error in fresh `composer.json` but was unsuccessful. I then tried deleting `./vendor/` as well as `composer.lock` and now `composer install` works. I will update if I discover anything else, but at least this seems to be a workaround...

**Comment by dapphp** [comment]
I had a similar issue recently and could not solve the issue by clearing composer cache and removing the vendor directory.

The error was:


    [Composer\DependencyResolver\SolverProblemsException]
    --
    Problem 1
    - Installation request for privatevendor/privatepackage1 3.* -> satisfiable by privatevendor/privatepackage1v3.3.0].
    - privatevendor/privatepackage1 v3.3.0 requires guzzlehttp/guzzle ^6.0 -> no matching package found.

We had "minimum-stability": "dev" in the composer config. Removing this and then re-running composer update solved the conflict issue. It had not been a problem before so I don't know if something with composer itself or packagist changed.

**Comment by gmponos** [comment]
As many of you mention you had issue and the solution was found from your composer configuration and it was unrelated with guzzle.

Guzzle 6.2.1 is up in packagist and it is still downloaded by few ppl.. https://packagist.org/packages/guzzlehttp/guzzle/stats#6.2.1
Same for 5.3.3 https://packagist.org/packages/guzzlehttp/guzzle/stats#5.3.3

Also this question seems better suited to other channels like stackoverflow..

**Comment by carloscarucce** [comment]
Sorry for the lateness guys.

@GrahamCampbell Even if I pasted composer.json content here, I think you couldn't reproduce because it has many private packages. Some of them also depends on guzzle. That's why I pasted only aws sdk error message.

@dapphp That is it:

Turns out someone inserted `"minimum-stability": "dev"` in our composer.json too.
We figured out when we ran update/install command with **--no-dev** and **--prefer-stable** flags
